Dmp docker (#25)
ualbertalib/DMP_roadmap: README.md

#### License
The DMP Roadmap project uses the <a href="./LICENSE.md">MIT License</a>.

# Docker

> This project comes with a docker setup to easily setup your own local development environment for jupiter in just a few steps.

Reviewer: ```suggestion
This project comes with a docker setup to easily setup your own local development environment for DMP assistant in just a few steps.
```

## Step 1: Make sure you have docker and docker-compose installed:

1. [Install Docker](https://docs.docker.com/engine/installation/) (Requires version 1.13.0+)
2. [Install Docker Compose](https://docs.docker.com/compose/install/) (Requires version 1.10.0+)

### Still need more help? Check out the following

#### OSX / Windows
- If you are on Mac, check out [Docker for Mac](https://docs.docker.com/docker-for-mac/)
- If you are on Windows, check out [Docker for Windows](https://docs.docker.com/docker-for-windows/)

These will install `docker`, `docker-compose`, and `docker-machine` on your machine.

#### Linux

Use your distribution's package manager to install `docker` and `docker-compose`.

## Step 2: Get DMP Roadmap source code

> Clone the Jupiter repository from github:

Reviewer: ```suggestion
Clone the DMP repository from github:
```

```shell
git clone git@github.com:ualbertalib/DMP_roadmap.git
cd DMP_roadmap
```

## Step 3: Start docker and docker compose

### For development environment
To start and setup your docker containers run:
```shell
docker-compose -f docker-compose.lightweight.yml up -d
```

> Now everything should be up and running. If you need seed to setup your database, then run the following command:

Reviewer: Might want to link to https://github.com/DMPRoadmap/roadmap/wiki/Installation#installation because there are some steps there that aren't laid out here.

```shell

> bundle exec rake db:setup

Reviewer: I get a warning at this stage:
> Devise.secret_key was not set. Please add the following to your Devise initializer:

I did `rails generate devise:install` and followed the instructions for setting the secret key.  It looks like the `config/initializers/devise.rb` is in `.gitignore` which might be part of this problem?

Reviewer: There's something up with the migrations.  `rake db:migration` has errors.

Reviewer: `rake db:setup` was also unsuccessful
```
NoMethodError: undefined method `branding' for #<Rails::Application::Configuration:0x00005639ca269698>
/home/pjenkins/.rvm/gems/ruby-2.5.5@DMP/gems/railties-4.2.11.1/lib/rails/railtie/configuration.rb:95:in `method_missing'
/home/pjenkins/Code/DMP_roadmap/lib/branding.rb:20:in `fetch'
/home/pjenkins/Code/DMP_roadmap/db/seeds.rb:224:in `<top (required)>'
```

Reviewer: `rake db:create && rake db:schema:load && rake db:seed` from the https://github.com/DMPRoadmap/roadmap/wiki/Installation#installation worked.

```

Then start your rails server run:
```shell

> bundle exec rails s

Reviewer: More errros :(
> DEPRECATION WARNING: You didn't set `secret_key_base`. Read the upgrade documentation to learn more about this new config option. (called from require at bin/rails:4)
Unexpected error while processing request: Missing `secret_token` and `secret_key_base` for 'development' environment, set these values in `config/secrets.yml`

 https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#config-secrets-yml

Reviewer: And more...
```
Webpacker::Manifest::MissingEntryError - Webpacker can't find application.js in /home/pjenkins/Code/DMP_roadmap/public/packs/manifest.json. Possible causes:
1. You want to set webpacker.yml value of compile to true for your environment
   unless you are using the `webpack -w` or the webpack-dev-server.
2. webpack has not yet re-run to reflect updates.
3. You have misconfigured Webpacker's config/webpacker.yml file.
4. Your webpack configuration is not creating a manifest.
Your manifest contains:
{
}
:
  app/views/layouts/application.html.erb:37:in `_app_views_layouts_application_html_erb__3917457177110087993_47426904055040'
```

```
### For UAT environment
To start your docker containers run:
```shell

> docker-compose -f docker-compose.yml up -d

Reviewer: docker-compose.yml is the default could just suggest
```suggestion
docker-compose up -d
```
